doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+
+![Switch-Access-Route](https://user-images.githubusercontent.com/61490193/223190966-7498a860-98d0-456f-8acb-241612a51790.png)
+
+
+#### Switch from access to trunk
+Switching from access to trunk will work without any error and untagged member added in the access mode is retained:
+
+![Switch-Access-Trunk](https://user-images.githubusercontent.com/61490193/223190991-aeef8fd6-e71a-4bbf-9862-42ea4613002a.png)
+
+To verify the port mode user can make of “show interfaces status” command
+
+
+#### Switching Ports & PortChannels mode from Trunk to Routed/Access
+
+#### Switch from trunk to routed
+Switch from trunk to routed user need to remove the untagged/tagged member(s) first otherwise, the command will give an error:
+
+![Switch-Trunk-Routed](https://user-images.githubusercontent.com/61490193/223191099-c037754d-c221-43b6-874e-6dbcd4eae639.png)
+
+
+#### Switch from trunk to access
+Switch from trunk to access, user need to remove the tagged members firstotherwise, the command will give an error:
+
+![Switch-Trunk-Access](https://user-images.githubusercontent.com/61490193/223191151-cea02ab6-7755-4c7c-994c-72bbbaeccc56.png)
+
+
+
+###  Examples/Usage of VLAN CLI Enchancemnts Command
+
+ #### Add multiple Vlan(s) using a single command:
+
+```
+  admin@sonic:~$ sudo config vlan add -m 100,200,300
+
+  Usage:  This command will add multiple comma separated vlan in a list
+ 
+ Example : The command will create the VLAN "Vlan 100, Vlan 200, Vlan 300" if these do not already exist.
+
+  ```
+  ![Vlan-Comma Separted](https://user-images.githubusercontent.com/61490193/223199523-c6e7ddb4-023a-4c9d-8a49-ec7d637bcd31.png)
+
+
+ ####  Add range of Vlan(s) using a single command:
+
+```
+  admin@sonic:~$ sudo config vlan add -m 10-20
+
+  Usage:  This command will add range of vlan in a list
+ 
+  Example : The command will create the VLAN "Vlan10 Vlan11, Vlan12, Vlan13, Vlan14, Vlan15, Vlan16, Vlan17, Vlan18, Vlan19, Vlan20" if these do not already exist.
+
+  ```
+![1](https://user-images.githubusercontent.com/61490193/223469654-38c17b89-cc07-4b5d-92b2-4e1dd19370de.png)
+
+  
+ The “-m” flag works the same way for deleting multiple VLANs as it does for adding vlans.
+
+
+ ####  Add all Vlan(s) as Vlan Member(s) using a single command:
+
+```
+  admin@sonic:~$ sudo config vlan member add all Ethernet20
+
+  Usage:  This command will add all existing vlan(s) as vlan member(s) 
+ 
+ Example : Suppose Vlan2, Vlan3, Vlan4, Vlan5, Vlan6, Vlan7, Vlan8, Vlan9 are existing Vlans. This command will add Ethernet20 as a tagged member of Vlan2, Vlan3, Vlan4, Vlan5, Vlan6, Vlan7, Vlan8, Vlan9
+ 
+   ***This command only work for trunk ports***
+
+  ```
+  
+![Vlan-MultipleAdd](https://user-images.githubusercontent.com/61490193/223199652-01b5ed35-15c0-480c-ad78-c78566db62df.png)
+
+
+ ####  Add all Vlan(s) as Vlan Member(s) excluding a range of vlans using a single command:
+
+  ```
+
+  admin@sonic:~$ sudo config vlan member add -m -e 12-17 PortChannel0003
```

Comment:
Is this correct? I thought -m is for range of VLANs... should it be add all -e 12-17 instead?

Reply:
Yes, this is correct. We will add more examples in this regard to avoid confusion.